rustls/src/server/tls13.rs

```diff
@@ -487,10 +487,9 @@ mod client_hello {
     ) -> Result<KeyScheduleHandshake, Error> {
         let mut extensions = Vec::new();
 
-        // Prepare key exchange
+        // Prepare key exchange; the caller ascertained that the `share.group` is supported
         let kx = kx::KeyExchange::choose(share.group, &config.kx_groups)
-            .and_then(kx::KeyExchange::start)
```

Comment:
I think this means the `start()` function can be removed from the trait? Since there are now no longer any calls to it.

Reply:
(nb. this is mostly a note to myself, as I work through ensuring these APIs can be used outside the crate.)

Reply:
Actually I was wrong about this -- there is one call, but I think one that can be refactored away.

Reply:
This is done; that commit could be squashed into earlier ones perhaps.

Reply:
Nice!